Library/Homebrew/keg.rb

```diff
@@ -377,6 +378,14 @@ def delete_pyc_files!
     find { |pn| pn.delete if pn.extname == ".pyc" }
   end
 
+  def cleanup
+    if linked?
+      opoo "Skipping (old) #{to_s} due to it being linked"
+    else
+      Homebrew.cleanup_path(self) { uninstall }
+    end
+  end
```

Comment:
Since this method is only used in `Formula#cleanup`, I would suggest to inline the code inside `Formula#cleanup`. Part of another reason to do so, it could cause confusion between `Keg#uninstall` and `Keg#cleanup` without reading the code.